### PR TITLE
schema docs correctly show required status

### PR DESCRIPTION
**Pull request by awgymer**
The code for `nf-core schema docs` assumed that `required` was a field on the `parameter` object in the `json`. It is actually an array at the level of the properties. Due to this `param.get('required')` was always false and auto-generated docs display every parameter as optional.

This PR fixes the code to check for the parameters inclusion in the `required` array.  
 
## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/schema.py [hunk 1]
```diff
@@ -513,12 +514,15 @@ def schema_to_markdown(self, columns):
...
                         out += f"| `{param.get('type', '')}` "
+                    elif column == "required":
+                        out += f"| {p_key in required or ''} "
                     else:
                         out += f"| {param.get(column, '')} "
                 out += "|\n"
 
         # Top-level ungrouped parameters
         if len(self.schema.get("properties", {})) > 0:
+            required = self.schema.get("required", [])
```
Comment (mashehu) [inline nf_core/schema.py:525]: Should this one not also be added to `out` below, the same as we above?

Comment (awgymer) [reply nf_core/schema.py:525]: D'oh

Comment (awgymer) [reply nf_core/schema.py]: I've refactored the table generating code which was the same for the if/else block into its own method. I realised that I hadn't captured the `<br>` changes from the previous PR in the top-level properties section either 🙈

**Review by mirpedrol — APPROVED** [review]
LGTM :)

**Pull request merged**